pkg/apis/monitoring/v1/types.go

```diff
@@ -426,6 +433,40 @@ type PrometheusRuleExcludeConfig struct {
 	RuleName string `json:"ruleName"`
 }
 
+// ObjectReference references a PodMonitor, ServiceMonitor, Probe or PrometheusRule object.
+type ObjectReference struct {
+	// Group of the referent. When not specified, it defaults to `monitoring.coreos.com`
+	// +optional
+	// +kubebuilder:default:="monitoring.coreos.com"
+	Group string `json:"group"`
+	// Resource of the referent.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=prometheusrules;servicemonitors;podmonitors;probes
+	Resource string `json:"resource"`
+	// Namespace of the referent.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Namespace string `json:"namespace"`
+	// Name of the referent. When not set, all resources are matched.
+	// +optional
+	Name string `json:"name,omitempty"`
+}
+
+func (obj *ObjectReference) GroupResource() schema.GroupResource {
+	return schema.GroupResource{
+		Resource: obj.Resource,
+		Group:    obj.GetGroup(),
+	}
+}
+
+func (obj *ObjectReference) GetGroup() string {
```

Comment:
You're right - the reason for defensive default is to ensure the tests do not sneak in an empty value (when crafting structs in tests we are not passing them through opneapi validation)

Reply:
could we make the method private since it's only used in this package?

```suggestion
// getGroup returns the group of the object.
// It is mostly needed for tests which don't create objects through the API and don't benefit from the default value.
func (obj *ObjectReference) getGroup() string {
```